src/components/form/form.tsx

```diff
@@ -118,13 +119,32 @@ const fields: Array<FieldInterface> = [
 
 export default function Form({
   isOpen,
-  onClose,
+  closeForm,
 }: FormProps): JSX.Element | null {
+  const [formError, setFormError] = useState<boolean>(false);
+  const [formSuccess, setFormSuccess] = useState<boolean>(false);
+
   async function onSubmit(form: FormValues): Promise<void> {
+    // clear out existing error state if present
+    if (formError) {
+      setFormError(false);
+    }
+
     const body = form as unknown as Registration; // converts required formValues arg type to required Registration type for POST
-    // Temporary Implementation
-    await uploadToAppServices(body);
-    return;
+
+    const req = await uploadToAppServices(body);
```

Comment:
Ah so `uploadToAppServices` should not be called from the browser. It requires an API key that we want to keep only on the server side. Rather than calling this directly, call the API route set up in the app at `/api/registration`. This will allow for us to act as a middleperson between the browser and the App Services function.

Reply:
got it, so within this function, I should do something like:

```
async function onSubmit() {
  // ...other local state stuff
  const req = await axios.post('/api/registration', body);
  const res = await req.json();

  if (res.ok) {
    // success stuff
  } else {
    // error stuff
  }
}

```

Reply:
Yeah exactly, you can probably just do a try/catch though since axios will resolve on 200-status responses.